### ISSUE TITLE
HookCallback hooks does not register LSTM for the text classifier 

### ISSUE THREAD

**Issue by vimarshc** [fastai/fastai#2385]
**Describe the bug**
The [HookCallBack](https://github.com/fastai/fastai/blob/master/fastai/callbacks/hooks.py#L62) to register a hook for each module does the following: 

    def on_train_begin(self, **kwargs):
        "Register the `Hooks` on `self.modules`."
        if not self.modules:
            self.modules = [m for m in flatten_model(self.learn.model)
                            if hasattr(m, 'weight')]
        self.hooks = Hooks(self.modules, self.hook)

However, the Pytorch LSTM does not have a `weight` parameter. 

So if I have a `text_classifier_learner` like: 

    >>> learn = text_classifier_learner(data_clas, AWD_LSTM,drop_mult=0.5,callback_fns=ActivationStats)

and then do 

    >>> modules = [m for m in flatten_model(learn.model)
                            if hasattr(m, 'weight')]
    >>>modules
    [Embedding(60000, 400, padding_idx=1),
    Embedding(60000, 400, padding_idx=1),
    BatchNorm1d(1200, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True),
    Linear(in_features=1200, out_features=50, bias=True),
    BatchNorm1d(50, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True),
    Linear(in_features=50, out_features=2, bias=True)]

**Expected behavior**

The HookCallback should be registering LSTM as well.

**Additional context**
 I think this could be fixed by checking is `m` is an instance on `torch.nn.Module`
I would like to take this up if this passes triage.

**Comment by sgugger** [comment]
Yes, this is a clunky check to make sure the model has parameters. This is corrected in v2 already, I'll push a fix for v1.

**Comment by sgugger** [comment]
Should be fixed now. Thanks for flagging!
